### PR TITLE
Hashan/simd optimization

### PR DESCRIPTION
**Pull request by xxhashanxx**
Complete SIMD optimization for delta

### CODE REVIEW

**Comment by xxhashanxx** [comment]
Seems DeltaDecodeSIMD performance worst than normal one. please check my implementation on  DeltaDecodeSIMD.

File: BENCHMARK.mb [hunk 1]
```diff
@@ -1,239 +1,253 @@
...
+BM_DeltaDecode/512              118 ns          118 ns      5795450 bytes_per_second=16.2234Gi/s
+BM_DeltaDecode/4096             958 ns          958 ns       745372 bytes_per_second=15.922Gi/s
+BM_DeltaDecode/32768           9547 ns         9548 ns        81424 bytes_per_second=12.7852Gi/s
+BM_DeltaDecode/65536          18500 ns        18501 ns        36067 bytes_per_second=13.1959Gi/s
+BM_DeltaDecodeSIMD/8           2.54 ns         2.54 ns    275182544 bytes_per_second=11.749Gi/s
+BM_DeltaDecodeSIMD/64          18.9 ns         18.9 ns     36562479 bytes_per_second=12.6207Gi/s
+BM_DeltaDecodeSIMD/512          227 ns          227 ns      3128738 bytes_per_second=8.39837Gi/s
+BM_DeltaDecodeSIMD/4096        2035 ns         2035 ns       339673 bytes_per_second=7.49841Gi/s
+BM_DeltaDecodeSIMD/32768      16525 ns        16526 ns        42024 bytes_per_second=7.38653Gi/s
+BM_DeltaDecodeSIMD/65536      33179 ns        33182 ns        20914 bytes_per_second=7.35773Gi/s
```
Comment (xxhashanxx) [inline BENCHMARK.mb]: DeltaDecodeSIMD take 33182ns when normal one take only 18501ns. I will try to upgrade this.
@gregulator Please check is there anything wrong with my implementation.

Comment (xxhashanxx) [reply BENCHMARK.mb]: improve performance with last commit, bellow you can see current result
BM_DeltaDecode/65536              18394 ns        18391 ns        37994 bytes_per_second=13.2747Gi/s
BM_DeltaDecodeSIMD/65536      15078 ns        15078 ns        46292 bytes_per_second=16.1919Gi/s

**Comment by xxhashanxx** [comment]
I have small issue with simple8b SIMD optimize version, In benchmarking when we use EncodeWords_SIMD in both ComputeSimple8bEncodeSize and  Simple8bEncode I get worst result but if I use it in only one I get very good result.
normal results:
----------------------------------------------------------------------------
Benchmark                                  Time             CPU   Iterations
----------------------------------------------------------------------------
BM_Simple8bEncodeUnsigned/1024        130431 ns       130426 ns         5333
BM_Simple8bEncodeUnsigned/4096        526451 ns       526428 ns         1287
BM_Simple8bEncodeUnsigned/32768      4178776 ns      4178709 ns          171
BM_Simple8bEncodeUnsigned/262144    33673445 ns     33673038 ns           21
BM_Simple8bEncodeUnsigned/1048576  134517145 ns    134514260 ns            5
BM_Simple8bEncodeUnsigned_BigO        128.29 N        128.29 N    
BM_Simple8bEncodeUnsigned_RMS              0 %             0 % 

use SIMD version on ComputeSimple8bEncodeSize :
----------------------------------------------------------------------------
Benchmark                                  Time             CPU   Iterations
----------------------------------------------------------------------------
BM_Simple8bEncodeUnsigned/1024         60182 ns        60177 ns        11821
BM_Simple8bEncodeUnsigned/4096        236149 ns       236149 ns         2932
BM_Simple8bEncodeUnsigned/32768      1872010 ns      1872020 ns          378
BM_Simple8bEncodeUnsigned/262144    15276841 ns     15276859 ns           46
BM_Simple8bEncodeUnsigned/1048576   60282577 ns     60281827 ns           11
BM_Simple8bEncodeUnsigned_BigO         57.54 N         57.54 N    
BM_Simple8bEncodeUnsigned_RMS              1 %             1 %   

use SIMD on both ComputeSimple8bEncodeSize and Simple8bEncode
Benchmark                                  Time             CPU   Iterations
----------------------------------------------------------------------------
BM_Simple8bEncodeUnsigned/1024        241773 ns       241768 ns         2860
BM_Simple8bEncodeUnsigned/4096        959099 ns       959100 ns          724
BM_Simple8bEncodeUnsigned/32768      7594551 ns      7594531 ns           90
BM_Simple8bEncodeUnsigned/262144    61829177 ns     61828709 ns           11
BM_Simple8bEncodeUnsigned/1048576  244287650 ns    244285867 ns            3
BM_Simple8bEncodeUnsigned_BigO        233.14 N        233.14 N    
BM_Simple8bEncodeUnsigned_RMS              1 %             1 %

**Review by gregulator — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**